test/comps/test_qb.cc

```diff
@@ -143,14 +133,12 @@ class TestQB : public ::testing::Test
         // TEST 3: Q'Q = I
         blas::syrk(Layout::ColMajor, Uplo::Upper, Op::Trans, k, m, 1.0, Q_dat, m, -1.0, Ident_dat, k);
 
-        // Get low-rank SVD
-        lapack::gesdd(Job::SomeVec, m, n, A_cpy_dat, m, s_dat, U_dat, m, VT_dat, n);
+        // buffer zero vector
         // buffer zero vector
         std::vector<T> z_buf(n, 0.0);
-        T* z_buf_dat = z_buf.data();
         // zero out the trailing singular values
-        blas::copy(n - k, z_buf_dat, 1, s_dat + k, 1);
-        RandLAPACK::util::diag(n, n, s, n, S);
+        blas::copy(n - k, z_buf.data(), 1, s_dat + k, 1);
```

Comment:
Defining ``z_buf`` and using it in this way is silly. Just use ``blas::scal(n-k, 0.0, s_dat, 1)`` or 
```
for (int i = 0; i < n - k; ++i)
    s_dat[i] = 0.0;
```

Reply:
True, that was just the approach i used when writing this thing last year. Can also just use std::fill.